### ISSUE TITLE
[bug] react/no-class-component Error boundary

### ISSUE THREAD

**Issue by Jack-Works** [Rel1cx/eslint-react#531]
### Describe the bug

components that use `getDerivedStateFromError` should not be warned because they have no migrations available.

### Platform and versions

```Text
"@eslint-react/eslint-plugin": "^1.5.13-next.1",
```
